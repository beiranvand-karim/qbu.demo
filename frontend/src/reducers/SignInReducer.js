import {
   CHECK_SIGN_IN_BEGIN,
   CHECK_SIGN_IN_SUCCESS,
   CHECK_SIGN_IN_ERROR
} from '../actions/SignInActions'

const initialState = {
   loading: false,
   error: null,
   loggedIn: false,
   token: null
};

export default function SignInReducer(state = initialState, action) {
   switch (action.type) {
      case CHECK_SIGN_IN_BEGIN:
         return {
            ...state,
            loading: true
         };

      case CHECK_SIGN_IN_SUCCESS:
         return {
            ...state,
            loading: false,
            loggedIn: true,
            token: action.payload
         };

      case CHECK_SIGN_IN_ERROR:
         return {
            ...state,
            loading: false,
            error: action.payload
         };

      default:
         return state;
   }
}
