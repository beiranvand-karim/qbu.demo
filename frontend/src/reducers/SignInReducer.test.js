/* global describe, it, expect */
import {
   CHECK_SIGN_IN_ERROR,
   CHECK_SIGN_IN_SUCCESS,
   CHECK_SIGN_IN_BEGIN
} from '../actions/SignInActions'
import SignInReducer from './SignInReducer'

describe('SignInReducer', () => {

   it('should matches the initial state', () => {
      const action = {type: 'DUMMY_ACTION'};
      expect(SignInReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the check sign in begin state', () => {
      const action = {type: CHECK_SIGN_IN_BEGIN};
      expect(SignInReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the check sign in successful state', () => {
      const action = {
         type: CHECK_SIGN_IN_SUCCESS,
         payload: 'ae6b174a-4259-4f02-a302-3dd3ca555414'
      };
      expect(SignInReducer(undefined, action)).toMatchSnapshot()
   });

   it('should matches the check sign in error state', () => {
      const action = {
         type: CHECK_SIGN_IN_ERROR,
         error: 'this is a error'
      };
      expect(SignInReducer(undefined, action)).toMatchSnapshot()
   });

});
