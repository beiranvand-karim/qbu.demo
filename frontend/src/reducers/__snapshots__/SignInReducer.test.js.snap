// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInReducer should matches the check sign in begin state 1`] = `
Object {
  "error": null,
  "loading": true,
  "loggedIn": false,
  "token": null,
}
`;

exports[`SignInReducer should matches the check sign in error state 1`] = `
Object {
  "error": undefined,
  "loading": false,
  "loggedIn": false,
  "token": null,
}
`;

exports[`SignInReducer should matches the check sign in successful state 1`] = `
Object {
  "error": null,
  "loading": false,
  "loggedIn": true,
  "token": "ae6b174a-4259-4f02-a302-3dd3ca555414",
}
`;

exports[`SignInReducer should matches the initial state 1`] = `
Object {
  "error": null,
  "loading": false,
  "loggedIn": false,
  "token": null,
}
`;
